### ISSUE TITLE
Bug CC0031 UseInvokeMethodToFireEventAnalyzer false positive in constructor

### ISSUE THREAD

**Issue by KristianWedberg** [code-cracker/code-cracker#926]
# Bug

CC0031 UseInvokeMethodToFireEventAnalyzer (incorrectly) suggests a fix when in a constructor in v1.0.3. In a regular method it works as documented:

````csharp
        public class WarnsInConstructor
        {
            public WarnsInConstructor(Action action)
            {
                if (action == null)
                    throw new ArgumentNullException();
                action();
            }

            public void RegularMethodWorksOK(Action action)
            {
                if (action == null)
                    throw new ArgumentNullException();
                action();
            }
        }
````

**Comment by giggio** [comment]
Bug confirmed.

**Comment by giggio** [comment]
Closed on #942.
